Falcon/Falcon/Rendering/PipeLine/Renderer.cpp

```diff
@@ -212,6 +216,24 @@ void Renderer::Draw()
 		}
 	}
 
+	uint32_t count = 0;
+	//Transparent meshes
+	for(auto it = distanceEntityMap.rbegin(); it != distanceEntityMap.rend(); it++)
+	{
+		RenderComponent* rc = m_entities->at(it->second)->GetComponent<RenderComponent>();
+		rc->m_mesh->AddWorldMatrix(m_entities->at(it->second)->GetTransform()->GetModel());
+		count++;
+
+		auto next = it;
+		next++;
+		if(m_entities->at(next->second)->GetComponent<RenderComponent>()->m_mesh != rc->m_mesh)
```

Comment:
this line is throwing an out of range exception. I believe its possible that its because when 'it' reaches the last element in your map and then you go next++ after that so you are going out of range of the map.